metrics/serve.go

```diff
@@ -103,19 +103,112 @@ func RunMetrics(metricCount, labelCount, seriesCount, metricLength, labelLength,
 
 	metricCycle := 0
 	seriesCycle := 0
-	registerMetrics(metricCount, metricLength, metricCycle, labelKeys)
-	cycleValues(labelKeys, labelValues, seriesCount, seriesCycle)
 	valueTick := time.NewTicker(time.Duration(valueInterval) * time.Second)
 	seriesTick := time.NewTicker(time.Duration(seriesInterval) * time.Second)
 	metricTick := time.NewTicker(time.Duration(metricInterval) * time.Second)
+	changeSeriesTick := time.NewTicker(time.Duration(seriesChangeInterval) * time.Second)
 	updateNotify := make(chan struct{}, 1)
 
+	var currentSeriesCount int
+
+	switch seriesOperationMode {
+	case "double-halve":
```

Comment:
it might be worth breaking the body of each `case` plus the default into 3 separate functions that can be called

Reply:
I've made the changes as you suggested👀